tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -812,15 +812,13 @@ export const downloadResources = async (args: downloadResourcesArgs): Promise<Do
     }
 
     case 'PREVIEW_TOPBAR':
-      await Promise.all([
-        page.locator(appBarDownloadFileButton).waitFor(),
-        page.locator(appBarContextMenu).click()
-      ])
+      const downloadButtonPromise = page.locator(appBarDownloadFileButton).waitFor()
+      await page.locator(appBarContextMenu).click()
+      await downloadButtonPromise
```

Comment:
what would be the difference to?
```suggestion
      await page.locator(appBarContextMenu).click()
      await page.locator(appBarDownloadFileButton).waitFor()
```

Reply:
We are looking for the button BEFORE the menu opens.
following docs
https://playwright.dev/docs/downloads#introduction

better do it like: 
1. Preparation 
2. Action
3. Wait

Reply:
let see how it works. I couldn't think of anything more interesting 😆